ocs_ci/deployment/deployment.py

```
@@ -3747,6 +3748,396 @@ def _patch_ramen_csv(self, csv_prefix, namespace, new_image):
+            return False
+
+        binary_path = f"{ramenctl_dir}/ramenctl"
+        if not os.path.exists(binary_path):
+            logger.error(
+                f"[ramenctl-e2e] Binary not found at {binary_path} after build"
+            )
+            return False
+
+        # Verify binary by running help command (ramenctl doesn't support --version)
+        verify_cmd = f"cd {ramenctl_dir} && ./ramenctl help"
+        logger.info(f"[ramenctl-e2e] Verifying binary with command: {verify_cmd}")
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)